[Loy Yoke Yue] iP

yokester98 opened the pull request:
# **Duke Chatbot**
Duke helps you to schedule your things to do. It's
- Easy to use
- Fast
- ~~Pay-to-win~~ Completely **_FREE_**
> "Why spend money on schedulers when Duke is free to use?" ~ CS student probably

## Simply follow the instructions below to get Duke:
1. Download it from [here](https://github.com/yokester98/ip).
2. Run the program.
3. Add your tasks.
4. Sit back and relax as Duke organises your schedule for you! 😄 

Current features include:
- [x] Manage tasks
- [ ] Reminders (coming soon!)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

yokester98 committed:
Level-1

yokester98 committed:
Level-2

yokester98 committed:
Level-3

yokester98 committed:
Level-3

yokester98 committed:
Level-4

yokester98 committed:
A-TextUiTesting

yokester98 committed:
Preparing for level-5

yokester98 committed:
Level-5

yokester98 committed:
Level-6

yokester98 committed:
Cleaning the code

yokester98 committed:
Added javadocs

yokester98 committed:
Add Level-7

yokester98 committed:
Add Level-7

yokester98 committed:
Add Level-8

yokester98 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

yokester98 committed:
Resolve conflicts

yokester98 committed:
Add Parser, Storage, Tasklist, Ui

yokester98 committed:
Add duke package

yokester98 committed:
Fix duke package

yokester98 committed:
Add JUnit Tests for Todo, Deadline, Event

yokester98 committed:
Add files for jar creation

yokester98 committed:
Add javadocs

yokester98 committed:
Add A-JavaDoc

yokester98 committed:
Edit code to follow coding standard

yokester98 committed:
Add Level-9

yokester98 committed:
Merge branch 'branch-A-CodingStandard'

yokester98 committed:
Merge branch 'branch-Level-9'

yokester98 committed:
Minor edits to Deadline and Parser

yokester98 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

yokester98 committed:
Add gradle to duke project

yokester98 committed:
Add CheckStyle

Korochii commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+        BufferedReader reader = new BufferedReader(fReader);
+        String line;
+        while ((line = reader.readLine()) != null) {
+            String[] strArr;
```
Might be a little nitpicking here but I think strArr should be in plural form since it is an array of Strings.

Korochii commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+            switch (strArr[0]) {
+            case "T":
+                task = new Todo(strArr[2]);
+                if (strArr[1].equals("1")) {
```
```suggestion
                if (strArr[1].equals("X")) {
```
I think there is a bug in which marked to-do tasks are being loaded as unmarked to-do tasks upon re-running Duke (which might be caused by this line)

yokester98 replied:
You are right! Thanks so much for noticing this! Fixed the issue in latest commit.

Korochii commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,86 @@
+     *
+     * @return  description of the task
+     */
+    public String getDescription() {
+        return this.description;
+    }
```
```suggestion
    public String getDescription() {
        return description;
    }
```
Is it necessary to include `this` here? (since removing it does not make the code ambiguous)

Korochii commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,86 @@
+    /**
+     * Sets the isDone attribute to true.
+     */
+    public void setDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Sets the isDone attribute to false.
+     */
+    public void setNotDone() {
+        this.isDone = false;
+    }
```
The implementation here works fine. However, my personal preference would be to merge these 2 methods into a single method that takes in a boolean value, since both methods are essentially doing the same thing (which is to set the value of isDone)

Korochii commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,120 @@
+     *
+     * @param type  The type of error to be thrown.
+     */
+    public void throwError(String type) {
+        System.out.println(LINE);
+        DukeException error;
+        if (type.equals("")) {
+            error = new DukeException();
+        } else {
+            error = new DukeException(type);
+        }
+        System.out.println(error);
+        System.out.println(LINE);
+    }
```
Interesting approach in handling exceptions! (which I have not thought of...)

Korochii reviewed (commented):
Overall, your code seems quite clear and readable, and had very little coding standard/quality violations as well (which was why I had to nitpick a few areas :sweat_smile:)

Good job and keep up the good work!

yokester98 committed:
Add GUI for Duke. Remove console outputs.

yokester98 committed:
Improve code quality

yokester98 committed:
Merge pull request #1 from yokester98/branch-A-CodeQuality

Improve code quality

yokester98 committed:
Add assertions to Parser

yokester98 committed:
Merge branch 'master' into branch-A-Assertions

yokester98 committed:
Merge pull request #2 from yokester98/branch-A-Assertions

Add assertions to Parser

yokester98 committed:
Make the command syntax more flexible.

Shorter aliases for keywords are now accepted as input. This was
done by providing more keywords that the Parser will accept.

yokester98 committed:
Package all Task classes into a single package.

Currently, all the classes are packaged into a single Duke package.

It needs to be further separated into their own packages to improve
code quality.

yokester98 committed:
Set theme jekyll-theme-cayman

yokester98 committed:
Add User Guide for Duke Task Manager.

README.md will be used for the user guide.

Ui.png will showcase what the Duke Task Manager looks like.

yokester98 committed:
Merge branch 'master' of https://github.com/yokester98/ip

yokester98 committed:
Fix javadocs for methods in Parser and Storage.

There were a few methods with incorrect javadoc format, this has
been amended in this commit.

yokester98 committed:
Change filepath that Duke stores and reads data from.

Previously, files were saved and read from user.home directory.
However, it is better to create a folder to contain the data.txt
where duke.jar is located.

yokester98 committed:
Improve README userguide
